content/ui-patterns/feature-onboarding.mdx

```diff
@@ -0,0 +1,271 @@
+
+- ✅ Use empty states or alert banners to call attention to a feature if users will discover your feature in order to start a task
+- ✅ Use teaching bubbles or other easily dismissable elements if users will discover your feature while they are completing another task
+
+## Begin a setup flow
+Use [empty states](https://primer.style/design/ui-patterns/empty-states), [teaching bubbles](https://docs.microsoft.com/en-us/javascript/api/office-ui-fabric-react/iteachingbubbleprops?view=office-ui-fabric-react-latest), or a [CTA on a marketing page](https://github.com/features/actions) on GitHub to begin a setup flow.
+
+![blank state](https://user-images.githubusercontent.com/47256527/88720353-97b62a00-d0d9-11ea-93d6-951c9da55d9c.png)
+
+![Actions example 1](https://user-images.githubusercontent.com/47256527/88720395-a56baf80-d0d9-11ea-9bd6-e98df7a9f27e.png)
+
+
+## Guide your user through their task
+Use a [form](https://primer.style/css/components/forms), [checklist](https://primer.style/css/components/forms), or other UI elements to help users complete their task. 
+
+![Checklist example](https://user-images.githubusercontent.com/47256527/88720479-c7fdc880-d0d9-11ea-9ac1-76b9defafe79.png)
+
+## Celebrate the finish
+Use a toast, flash alert, or other message to confirm the user has completed their task. Message should be short and brief.
+
+![Do](https://user-images.githubusercontent.com/47256527/88720500-d3e98a80-d0d9-11ea-8943-43e6e04431b3.png)
+
+ 
+# Teaching bubble
+Teaching bubbles are a special [popover]() that calls attention to a feature in a specific part of the page. Generally, features highlighted by teaching bubbles should be used to educate the user and enriching to the user’s task at hand. Consider the page the user is currently viewing and what jobs-to-be-done they may be performing.
+
+|Pattern |Quickstart| Top user benefits|
+|:--|:--|:--|
+|Whent to use |<li>Call attention to new or important features</li><li>Invoked by the system, like a new capability being available</li>|The highlighted feature extends or improves the user's likely task at hand|
+|When not to use |<li>Don't point to hidden elements</li><li>If the user cannot immediately interact and benefit from the feature highlighted</li>|Do not distract the user from completing their task.|
+
+## General guidelines
+Refer to the [Primer popover component](https://primer.style/css/components/popover) for more detailed implementation guidelines.
+
+```html live title="Basic"
+<div class="Popover right-0 left-0 position-relative">
+  <div class="Popover-message text-left p-4 mt-2 mx-auto Box box-shadow-large">
+    <h4 class="mb-2">Teaching bubble heading</h4>
+    <p>Use this as a teaching moment that may enrich a user’s experience. Keep message to a maximum of 160 characters.</p>
+    <button class="btn btn-sm btn-outline mt-2" type="button">Primary action</button>
+    <button class="btn btn-invisible ml-2">Ok, dismiss</button>
+  </div>
+</div>
+```
```

Comment:
Do we need to show this code here?

Reply:
Maybe we can lead with the list in this section, and move the implementation note to after the list?

Reply:
It was just that our track record shows popovers are implemented with slightly different spacing and typography each time...

We've created an [onboarding component sticker sheet](https://www.figma.com/file/FJtZyBeBfwDkByHPfDSN6k/Feature-onboarding?node-id=857%3A6) so the inconsistencies may fix with a resource like? 🤞